lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1466,10 +1466,14 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
       })
 
       // Handle memories.
-      .Case<MemOp>([&](auto op) {
+      .Case<MemOp>([&](MemOp op) {
         // Create constraint variables for all ports.
-        for (auto result : op.getResults())
-          declareVars(result, op.getLoc());
+        unsigned nonDebugPort = 0;
+        for (const auto &result : llvm::enumerate(op.getResults())) {
+          declareVars(result.value(), op.getLoc());
+          if (!result.value().getType().cast<FIRRTLType>().isa<RefType>())
+            nonDebugPort = result.index();
```

Comment:
~~What happens if there are multiple debug ports, or the debug ports are not all at the beginning? Should we error out if that's the case?~~

Reading on a bit it looks like you just need *any* non-debug port in order to have something to unify all other non-debug ports to. That makes total sense!

Reply:
Right, any non-debug port will be used to infer the data type.